### PR TITLE
Add Typescript support

### PR DESCRIPTION
**Pull request by Betree**
Resolve https://github.com/opencollective/opencollective/issues/5710
Moved from https://github.com/opencollective/opencollective-frontend/pull/8009

**Notes**
- All files in `lib/graphql/types/v2/` are auto-generated though the usual `npm run graphql:update` command. It's basically a converstion of the GraphQL schema V2 to Typescript types.
- Some technical constraints are making it difficult to extract types from multiple APIs, I've chosen to ignore V1 entirely for this 1st release.

**Some examples**
- Migrate a styled component with styled-system: https://github.com/opencollective/opencollective-frontend/pull/8156/files#diff-a0646825e3169cb68522099621c64159b0b49d9f1a35940623efc393814872f1
- Typing a React context: https://github.com/opencollective/opencollective-frontend/pull/8156/files#diff-54d03659385a3d49b1f84123c9e63a54259de7378690c23fa87a49219454a64b

**Impact on bundle size**

Nothing alarming.

> `/root-actions`: 738 kB -> 744 kB
> First Load JS shared by all              490 kB -> 489 kB

### CODE REVIEW

File: components/CollectivePickerAsync.js [hunk 1]
```diff
@@ -106,18 +106,18 @@ const getPlaceholder = (intl, types) => {
...
+  hostCollectiveIds = undefined,
+  parentCollectiveIds = undefined,
+  preload = false,
+  filterResults = undefined,
+  searchQuery = collectivePickerSearchQuery,
+  invitable = false,
+  emptyCustomOptions = [],
+  noCache = false,
+  isLoading = false,
+  skipGuests = true,
```
Comment (Betree) [inline components/CollectivePickerAsync.js:120]: Here typescript needed a way to understand that these props are optional. I did not want to convert the entire thing to TS, so I just moved the `defaultProps` to default values, which is anyway the [recommended](https://twitter.com/dan_abramov/status/1133878326358171650) way to do things.

File: lib/custom_typings/images.d.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+declare module '*.svg' {
+  const value: any;
+  export = value;
+}
+
+declare module '*.gif' {
+  const value: any;
+  export = value;
+}
```
Comment (Betree) [inline lib/custom_typings/images.d.ts:19]: This part was required for TS to understand that webpack can import image files directly (and that they shouldn't be parsed like regular code files)

**Review by znarf — APPROVED** [review]
Looking good to me!

**Review by hdiniz — APPROVED** [review]
LGTM!

**Comment by Betree** [comment]
Thanks for the reviews! I'll try to merge it tomorrow

**Pull request merged**